lambdas/tests/unit/services/test_bulk_upload_metadata_service.py

```diff
@@ -0,0 +1,241 @@
+        "S3:HeadObject",
+    )
+    expected_err_msg = (
+        "An error occurred (403) when calling the S3:HeadObject operation: Forbidden"
+    )
+
+    service = BulkUploadMetadataService()
+    with pytest.raises(ClientError) as e:
+        service.process_metadata(metadata_filename)
+
+    assert str(e.value) == expected_err_msg
+    mock_sqs_service.send_message_with_nhs_number_attr_fifo.assert_not_called()
+
+
+def test_process_metadata_raise_validation_error_when_metadata_csv_is_invalid(
+    set_env, metadata_filename, mock_sqs_service, mock_download_metadata_from_s3
+):
+    service = BulkUploadMetadataService()
+
+    for invalid_csv_file in MOCK_INVALID_METADATA_CSV_FILES:
+        mock_download_metadata_from_s3.return_value = invalid_csv_file
+
+        with pytest.raises(ValidationError):
+            service.process_metadata(metadata_filename)
+
+        mock_sqs_service.send_message_with_nhs_number_attr_fifo.assert_not_called()
+
+
+def test_process_metadata_raise_client_error_when_failed_to_send_message_to_sqs(
+    set_env, metadata_filename, mock_s3_service, mock_sqs_service, mock_tempfile
+):
+    mock_client_error = ClientError(
+        {
+            "Error": {
+                "Code": "AWS.SimpleQueueService.NonExistentQueue",
+                "Message": "The specified queue does not exist",
+            }
+        },
+        "SendMessage",
+    )
+    mock_sqs_service.send_message_with_nhs_number_attr_fifo.side_effect = (
+        mock_client_error
+    )
+    expected_err_msg = (
+        "An error occurred (AWS.SimpleQueueService.NonExistentQueue) when calling the SendMessage operation:"
+        " The specified queue does not exist"
+    )
+
+    service = BulkUploadMetadataService()
+    with pytest.raises(ClientError) as e:
+        service.process_metadata(metadata_filename)
+
+    assert str(e.value) == expected_err_msg
+
+
+def test_download_metadata_from_s3(
+    set_env, metadata_filename, mock_s3_service, mock_tempfile
+):
+    service = BulkUploadMetadataService()
+    actual = service.download_metadata_from_s3(metadata_filename)
+    expected = MOCK_METADATA_CSV
+
+    mock_s3_service.download_file.assert_called_with(
+        s3_bucket_name=MOCK_LG_STAGING_STORE_BUCKET,
+        file_key=metadata_filename,
+        download_path=f"{MOCK_TEMP_FOLDER}/{metadata_filename}",
+    )
+    assert actual == expected
+
+
+def test_download_metadata_from_s3_raise_error_when_failed_to_download(
+    set_env, metadata_filename, mock_s3_service, mock_tempfile
+):
+    service = BulkUploadMetadataService()
+    mock_s3_service.download_file.side_effect = ClientError(
+        {"Error": {"Code": "500", "Message": "file not exist in bucket"}},
+        "s3_get_object",
+    )
+
+    with pytest.raises(ClientError):
+        service.download_metadata_from_s3(metadata_filename)
+
+
+def test_csv_to_staging_metadata(set_env):
+    service = BulkUploadMetadataService()
```

Comment:
Maybe move the service to a fixture?

Reply:
Agree. I have replaced all these calls with a `metadata_service` fixture.